### PR TITLE
fix: Add icon granularity to HTML code generation

### PR DESCRIPTION
**Pull request by Voltra**
## Changes
- Make HTML codegen conditional according to granularity of icon sets
- Refactor HTML codegen (reducing boilerplate and copy/paste issues)

## Why
From #312 it has been brought to my attention that HTML codegen was not affected by the newly added granularity.
This aims to only generate HTML for icons that were specified.

## How
With a few helpers, check if the options for the working icon set is an array and if it is, only generate the selected options. If it's not an array, proceed as usual.

## Feedback
It modified a lot of the HTML codegen structure to allow for better maintainability and flexibility (e.g. adding new icons). This also clears a lot of "code smells" were a lot of lines were simply duplicates with small adjustments. Also brought to the light that there were a lot of strange decision regarding whitespaces (especially between HTML attributes for some reason) as well as a bad habit of not autoclosing tags (i.e. leaving them open as if they were opening tags).

You can take a look at [config/html.js from line 40 to line 71](https://github.com/itgalaxy/favicons/blob/v6.2.0/src/config/html.js#L49), especially the `href` attributes and end of tags to know what I'm talking about.

## Nota Bene
I will not in any way shape or form alter the tests and/or fixtures/snapshots that rely on mentioned strange decisions.
As such, some tests fail (due to missing whitespaces).

Improvements in code quality makes it trivial to implement #289 on top of it.

### CODE REVIEW

**Comment by Voltra** [comment]
I will, as soon as all the test stop relying on arbitrary whitespace constraints.

**Comment by evilebottnawi** [comment]
@Voltra What is the next step for us?

**Comment by Voltra** [comment]
Snapshots/fixtures should be rewritten to avoid the use of specific amounts of whitespace per item (only 1 per attribute). I also recommend the use of self-closing tags (i.e. `<meta/>` instead of `<meta>`). These are the cause of all failures. In an ideal scenario, tests would diff the DOM structure instead of the actual string.

Once all the whitespace issues are gone, I can proceed with tests for granularity on the HTML code generation level.

**Comment by evilebottnawi** [comment]
> Once all the whitespace issues are gone, I can proceed with tests for granularity on the HTML code generation level.

Do you want to send a fix?

File: test/snapshots/array.test.js.md [hunk 1]
```diff
@@ -123,34 +123,34 @@ Generated by [AVA](https://avajs.dev).
...
+        '<link rel="apple-touch-icon" sizes="72x72" href="/apple-touch-icon-72x72.png"/>',
+        '<link rel="apple-touch-icon" sizes="76x76" href="/apple-touch-icon-76x76.png"/>',
+        '<link rel="apple-touch-icon" sizes="114x114" href="/apple-touch-icon-114x114.png"/>',
+        '<link rel="apple-touch-icon" sizes="120x120" href="/apple-touch-icon-120x120.png"/>',
+        '<link rel="apple-touch-icon" sizes="144x144" href="/apple-touch-icon-144x144.png"/>',
+        '<link rel="apple-touch-icon" sizes="152x152" href="/apple-touch-icon-152x152.png"/>',
+        '<link rel="apple-touch-icon" sizes="167x167" href="/apple-touch-icon-167x167.png"/>',
+        '<link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon-180x180.png"/>',
+        '<link rel="apple-touch-icon" sizes="1024x1024" href="/apple-touch-icon-1024x1024.png"/>',
+        '<meta name="apple-mobile-web-app-capable" content="yes"/>',
```
Comment (evilebottnawi) [inline test/snapshots/array.test.js.md:144]: I think we should avoid `/` here and introduce the `html5` option, but we can do it in separate PR

Comment (Voltra) [reply test/snapshots/array.test.js.md:144]: Although they are considered useless as per the HTML5 standard, they still carry the meaning of being closed immediately with no content

Comment (evilebottnawi) [reply test/snapshots/array.test.js.md:144]: Let's avoid them anyway, I'm sure users will create a problem about this.

**Comment by Voltra** [comment]
The tests pass for the HTML generation

**Comment by evilebottnawi** [comment]
@Voltra What is the next?

**Comment by dmnsgn** [comment]
@Voltra has the html code generation been merged and published for? Latest release seem to be "on 21 Jul 2020" so I would assume that no because you would like to fix manifest generation as well?
